packages/flutter/lib/src/rendering/grid.dart

```diff
@@ -435,8 +473,8 @@ class RenderGrid extends RenderVirtualViewport<GridParentData> {
       double tileTop = _specification.rowOffsets[placement.row] + gridTopPadding;
```

Comment:
the grid left padding and grid top padding should be added before the column/row offsets, since they come first. Adding the numbers "in order" makes this easier to understand.